### PR TITLE
PLAT-121001: Agate Docs: Write missing documentation for components

### PR DESCRIPTION
**Pull request by paul-beldean-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
- Some components didn't have documentation

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Added missing documentation for components

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-121001

### Comments

Enact-DCO-1.0-Signed-off-by: Paul Beldean paul.beldean@lgepartner.com

### CODE REVIEW

File: FullscreenPopup/FullscreenPopup.js [hunk 1]
```diff
@@ -37,12 +37,68 @@ import componentCss from './FullscreenPopup.module.less';
...
+		/**
+		 * Type of popup.
+		 *
+		 * * Supported types are: `'slide'`, `'clip'`, and `'fade'`.
+		 *
+		 * @type {('slide'|'clip'|'fade')}
+		 * @default 'slide'
+		 * @public
+		 */
 		type: PropTypes.string
```
Comment (daniel-stoian-lgp) [inline FullscreenPopup/FullscreenPopup.js:102]:
```suggestion
		type: PropTypes.oneOf(['slide', 'clip', 'fade']),
```

File: Panels/TabbedPanels.js [hunk 2]
```diff
@@ -39,19 +39,123 @@ import componentCss from './TabbedPanels.module.less';
 const TabbedPanelsBase = kind({
 	name: 'TabbedPanels',
 	propTypes: /** @lends agate/Panels.TabbedPanels.prototype */ {
+		/**
+		* Components to be included after the tabs.
```
Comment (daniel-stoian-lgp) [inline Panels/TabbedPanels.js:43]: maybe 'Nodes' is a better word, instead of 'Components'

File: Panels/TabbedPanels.js [hunk 3]
```diff
@@ -39,19 +39,123 @@ import componentCss from './TabbedPanels.module.less';
...
+		 * viewport.
+		 *
+		 * @see {@link ui/ViewManager.SlideRightArranger}
+		 * @type {ui/ViewManager.Arranger}
+		 * @public
+		 */
 		arranger: shape,
+
+		/**
+		* Components to be included before the tabs.
```
Comment (daniel-stoian-lgp) [inline Panels/TabbedPanels.js:61]: here also. Nodes instead of Components

File: Panels/TabbedPanels.js [hunk 4]
```diff
@@ -39,19 +39,123 @@ import componentCss from './TabbedPanels.module.less';
...
+		/**
+		 * Index of the active panel
+		 *
+		 * @type {Number}
+		 * @default 0
+		 */
 		index: PropTypes.number,
+
+		/**
+		 * Disable tabbed panels transitions.
```
Comment (daniel-stoian-lgp) [inline Panels/TabbedPanels.js:103]:
```suggestion
		 * Disables tabbed panels transition animations.
```

**Review by daniel-stoian-lgp — COMMENTED** [review]
Congratulations for your first PR!

Just a few adjustments are required

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

File: Drawer/Drawer.js [hunk 5]
```diff
@@ -31,17 +31,62 @@ import componentCss from './Drawer.module.less';
...
+		/**
+		 * Header for the drawer.
+		 *
+		 * @type {Node}
+		 * @public
+		 */
 		header: PropTypes.node,
+
+		/**
+		 * Disable drawer transitions.
```
Comment (adrian-cocoara-lgp) [inline Drawer/Drawer.js:61]:
```suggestion
		 * Disables drawer transition animation.
```

File: PopupMenu/PopupMenu.js [hunk 6]
```diff
@@ -36,13 +36,67 @@ import componentCss from './PopupMenu.module.less';
...
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal elements and states of this component.
+		 *
+		 * @type {Object}
+		 * @public
+		 */
 		css: PropTypes.object,
+
+		/**
+		 * Disable popup menu transitions.
```
Comment (adrian-cocoara-lgp) [inline PopupMenu/PopupMenu.js:67]:
```suggestion
		 * Disables popup menu transition animation.
```

File: SliderButton/SliderButton.js [hunk 7]
```diff
@@ -57,10 +79,32 @@ const SliderKnob = kind({
...
+		 *
+		 * @type {Object}
+		 * @public
+		 */
 		css: PropTypes.object,
+
+		/**
+		 * Value of SliderProgress.
+		 *
+		 * @type {String[]}
```
Comment (adrian-cocoara-lgp) [inline SliderButton/SliderButton.js:105]:
```suggestion
		 * @type {String}
```

**Comment by jeongran1230** [comment]
In Agate Drawer.js, There is no description about `orientation` property. If absent, please add it.
```
defaultProps: {
   ...
  orientation: 'vertical'
}
```

**Comment by jeongran1230** [comment]
`defaultProp` does not exist for `IncrementSliderButtonBase` component
```
defaultProps: 
    orientation: 'horizontal',
    role: 'list',
    size: 'large'
```

**Comment by jeongran1230** [comment]
'closeButtonAriaLabel' property should be added in defaultProp object in TabbedPanelBase component (TabbedPanel.js)
```
defaultProps: {
+   closeButtonAriaLabel: 'Exit app'
    index:0
```

**Comment by paul-beldean-lgp** [comment]
> In Agate Drawer.js, There is no description about `orientation` property. If absent, please add it.
> 
> ```
> defaultProps: {
>    ...
>   orientation: 'vertical'
> }
> ```

The drawer doesn't seem to look ok with orientation="horizontal" so I think we could remove it from defaultProps and add it as a prop for the Layout inside the render. Would that be ok?

**Comment by jeongran1230** [comment]
> The drawer doesn't seem to look ok with orientation="horizontal" so I think we could remove it from defaultProps and add it as a prop for the Layout inside the render. Would that be ok?

I understood what you intended. I think it is more clear to define `orientation` as private prop by using @private tag like below.
```
propTypes: /** @lends agate/Drawer.DrawerBase.prototype */ {
   ...
         /**
		 * Orientation description..
		 *
		 * @type {String}
		 * @default 'vertical'
		 * @private
		 */
		orientation: PropTypes.oneOf(['horizontal', 'vertical']),
```

**Comment by jeongran1230** [comment]
I found there is no description for props in `TabGroupBase` class. Is it missed or having reason?

**Comment by jeongran1230** [comment]
Please add the following code: 
```--- a/Drawer/Drawer.js
+++ b/Drawer/Drawer.js
@@ -25,6 +25,7 @@ import componentCss from './Drawer.module.less';
  *
  * @class DrawerBase
  * @memberof agate/Drawer
+ * @extends ui/Transition.Transition
  * @ui
  * @public
  */
diff --git a/IncrementSlider/IncrementSliderButton.js b/IncrementSlider/IncrementSliderButton.js
index b0e06b3..a951257 100644
--- a/IncrementSlider/IncrementSliderButton.js
+++ b/IncrementSlider/IncrementSliderButton.js
@@ -14,6 +14,7 @@ import componentCss from './IncrementSliderButton.module.less';
  *
  * @class IncrementSliderButton
  * @memberof agate/IncrementSlider
+ * @extends agate/Button
  * @ui
  * @private
  */
```

**Comment by paul-beldean-lgp** [comment]
> I found there is no description for props in `TabGroupBase` class. Is it missed or having reason?

It was added in this pull request that was just merged: https://github.com/enactjs/agate/pull/488

**Comment by jeongran1230** [comment]
Congratulations your 1st PR.

**Comment by jeongran1230** [comment]
LGTM

**Comment by seunghoh** [comment]
@stanca-pop-lgp You have one pending review. Do you need more time to review?

**Review by seunghoh — CHANGES_REQUESTED** [review]
@paul-beldean-lgp Would you fix parse errors? There are lots of parsing errors when I try `npm run parse -- --extra-repos enactjs/agate#feature/PLAT-121001`

**Review by seunghoh — APPROVED** [review]

**Pull request merged**